test/test_video_gpu_decoder.py

```diff
@@ -31,10 +31,10 @@ def test_frame_reading(self):
             decoder = VideoReader(full_path, device="cuda:0")
             with av.open(full_path) as container:
                 for av_frame in container.decode(container.streams.video[0]):
-                    av_frames = torch.tensor(av_frame.to_ndarray().flatten())
+                    av_frames = torch.tensor(av_frame.to_rgb(src_colorspace="ITU709").to_ndarray())
                     vision_frames = next(decoder)["data"]
-                    mean_delta = torch.mean(torch.abs(av_frames.float() - decoder._reformat(vision_frames).float()))
-                    assert mean_delta < 0.1
+                    mean_delta = torch.mean(torch.abs(av_frames.float() - vision_frames.cpu().float()))
+                    assert mean_delta < 0.75
```

Comment:
I generally dislike performing the comparisons as the mean over the whole tensor because it can completely allow for having a whole section of the image to be all black / white / random (which is btw an existing issue with the current CPU video reader cc @bjuncek )

I would add a TODO here to try to use a different function for comparing equality here.

Reply:
Do you have a recommendation of what to use instead of `torch.mean()`? I see mean() being used at other places as well, for example: https://github.com/pytorch/vision/blob/5e56575e688a85a3bc9dc3c97934dd864b65ce47/test/test_image.py#L366
I know numpy has [assert_array_almost_equal](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_almost_equal.html), which I have used in the past to compare arrays. I guess `torch.allclose()` is an option.

Reply:
So, my preference would be to use `(a - b).abs().max() < threshold`. But looks like this doesn't work already in the CPU video decoder because of the black bands that appear sometimes (cc @bjuncek ), but ideally this is the type of things I would consider.

But this can be done in the future, just leave a TODO

Reply:
https://github.com/pytorch/vision/issues/5216